app/Http/Controllers/Api/V1/ProductController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Http\Controllers\Controller;
+use App\Services\ProductService;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+
+class ProductController extends Controller
+{
+    /**
+     * @var ProductService
+     */
+    private ProductService $productService;
+
+    /**
+     * @param ProductService $productService
+     */
+    public function __construct(ProductService $productService)
+    {
+        $this->productService = $productService;
+    }
```

Comment:
You can use php8.0 constructor promotion to avoid some boilerplate  https://stitcher.io/blog/constructor-promotion-in-php-8
```suggestion
    public function __construct(private ProductService $productService)
    {
    }
```